### PR TITLE
Updating the examples

### PR DESCRIPTION
**Pull request by Poolitzer**
### Checklist for PRs

- [ ] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [ ] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs

### If the PR contains API changes (otherwise, you can delete this passage)

* New classes:
    - [ ] Added `self._id_attrs` and corresponding documentation
    - [ ] `__init__` accepts `**_kwargs`
    
* Added new shortcuts:
    - [ ] In `Chat` & `User` for all methods that accept `chat/user_id`
    - [ ] In `Message` for all methods that accept `chat_id` and `message_id`
    - [ ] For new `Message` shortcuts: Added `quote` argument if methods accepts `reply_to_message_id`
    - [ ] In `CallbackQuery` for all methods that accept either `chat_id` and `message_id` or `inline_message_id`
    
* If relevant:

    - [ ] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [ ] Added new handlers for new update types
    - [ ] Added new filters for new message (sub)types
    - [ ] Added or updated documentation for the changed class(es) and/or method(s)
    - [ ] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`
    - [ ] Added logic for arbitrary callback data in `tg.ext.Bot` for new methods that either accept a `reply_markup` in some form or have a return type that is/contains `telegram.Message`

### CODE REVIEW

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the PR! changes themself look good except for the pre-commit errors. would you mind adding this section back to the pre-commit config?

https://github.com/python-telegram-bot/python-telegram-bot/blob/d117d21fffbb844b565f6cb9979c1c6afe601bc1/.pre-commit-config.yaml#L48-L59

I had removed it temporarily while working on the asyncio changes …

**Comment by Poolitzer** [comment]
I added the config and fixed the black thing. Black itself has currently an import issue in the pre-commit and pylint complains locally, so I forced git to commit anyway and am interested in how it runs here.

**Comment by Poolitzer** [comment]
I found the issue with black psf/black#2964, we either need to pin the click dependency to a lower version or upgrade the black version, what would you prefer and should I squeeze it in here or in a second PR?

**Comment by harshil21** [comment]
@Poolitzer We&#x27;re updating dependencies in <a href="https://github.com/python-telegram-bot/python-telegram-bot/pull/2925">#2925</a> so imo no need to do it here

<a href="https://github.com/python-telegram-bot/python-telegram-bot/pull/2925">PR #2925: Update Test Dependencies versions by harshil21</a>

**Comment by Bibo-Joshi** [comment]
@harshil21 I&#x27;m okay with doing it here/in asyncio as the CI currently fails otherwise. #2925 is due only after asyncio

**Comment by Poolitzer** [comment]
@Bibo-Joshi So to be less disruptive, I could hardpin the dependency, and we remove the hardpin and upgrade the version in #2925

**Review by Bibo-Joshi — APPROVED** [review]
The changes themself LGTM 👍 did you by any chance already test the examples? if not: do you want to before we merge or do we just do that later?

**Comment by Poolitzer** [comment]
@Bibo-Joshi I didn&#x27;t test it, I think that makes most sense before the first release.

File: examples/inlinebot.py [hunk 1]
```diff
@@ -56,14 +56,14 @@ async def inlinequery(update: Update, context: CallbackContext.DEFAULT_TYPE) ->
...
-                f"*{escape_markdown(query)}*", parse_mode=ParseMode.MARKDOWN
+                f"*{escape(query)}*", parse_mode=ParseMode.HTML
             ),
         ),
         InlineQueryResultArticle(
             id=str(uuid4()),
             title="Italic",
             input_message_content=InputTextMessageContent(
-                f"_{escape_markdown(query)}_", parse_mode=ParseMode.MARKDOWN
+                f"_{escape(query)}_", parse_mode=ParseMode.MARKDOWN
```
Comment (harshil21) [inline examples/inlinebot.py:66]:
```suggestion
                f"{escape(query)}", parse_mode=ParseMode.HTML
```

Comment (Poolitzer) [reply examples/inlinebot.py:66]: wupps, good catch

**Comment by harshil21** [comment]
Only thing I didn't run was passport, payment, and chatmemberbot. Rest all are working fine! Asyncio runs smooooth!

**Review by harshil21 — APPROVED** [review]

File: examples/rawapibot.py [hunk 2]
```diff
@@ -10,51 +10,56 @@
...
+                update_id = await echo(bot, update_id)
+            except NetworkError:
+                await asyncio.sleep(1)
+            except Forbidden:
+                # The user has removed or blocked the bot.
+                update_id += 1
 
 
-async def echo(bot: telegram.Bot) -> None:
+async def echo(bot: Bot, update_id: int) -> None:
```
Comment (Bibo-Joshi) [inline examples/rawapibot.py:45]:
```suggestion
async def echo(bot: Bot, update_id: int) -> int:
```

File: examples/rawapibot.py [hunk 3]
```diff
@@ -10,51 +10,56 @@
...
+    for update in updates:
+        next_update_id = update.update_id + 1
 
         # your bot can receive updates without messages
         # and not all messages contain text
         if update.message and update.message.text:
             # Reply to the message
+            logger.info("Found message %s!", update.message.text)
             await update.message.reply_text(update.message.text)
+        return next_update_id
```
Comment (Bibo-Joshi) [inline examples/rawapibot.py:58]:
```suggestion
        return next_update_id
    return update_id
```

File: examples/rawapibot.py (hunk 2 above)
Comment (harshil21) [reply examples/rawapibot.py:45]: hm weird mypy didn't catch that

**Comment by Poolitzer** [comment]
looks good to me

**Comment by Bibo-Joshi** [comment]
Tested chatmemberbot. Runs fine except for a but in `CMU.difference` that I found (see #2947), but that's not due to the example.

**Comment by Bibo-Joshi** [comment]
Tested passport & payment example. For passports, I adjusted the html file a bit and found #253 - which makes me want to stabilize `Updater` a bit more, but that's not for this PR.

If CI runs through & @harshil21 and @Poolitzer don't have anything to add, we can merge :)

**Comment by harshil21** [comment]
All good from my side

**Comment by Poolitzer** [comment]
Looks good to me as well

**Pull request merged**